client/core/core.go

```diff
@@ -2173,52 +2173,68 @@ func (c *Core) ToggleWalletStatus(assetID uint32, disable bool) error {
 		return nil
 	}
 
+	// If this wallet is a parent, disable/enable all token wallets.
+	affectedWallets := []*xcWallet{wallet}
 	if disable {
-		// Ensure wallet is not a parent of an enabled token wallet.
+		// Ensure wallet is not a parent of an enabled token wallet with active
+		// orders.
 		if assetInfo := asset.Asset(assetID); assetInfo != nil {
 			for id := range assetInfo.Tokens {
 				if wallet, exists := c.wallet(id); exists && !wallet.isDisabled() {
-					return fmt.Errorf("%s wallet has an enabled %s token wallet", unbip(assetID), unbip(id))
+					if c.assetHasActiveOrders(wallet.AssetID) {
+						return newError(activeOrdersErr, "active orders for %v", unbip(wallet.AssetID))
+					}
+					affectedWallets = append(affectedWallets, wallet)
 				}
 			}
 		}
 
 		if c.assetHasActiveOrders(assetID) {
 			return newError(activeOrdersErr, "active orders for %v", unbip(assetID))
 		}
+
+		// Ensure wallet is not an active bond asset wallet. This check will
+		// cover for token wallets if this wallet is a parent.
 		if c.isActiveBondAsset(assetID, true) {
 			return newError(bondAssetErr, "%v is an active bond asset wallet", unbip(assetID))
 		}
 
-		if wallet.connected() {
-			wallet.Disconnect() // before disable or it refuses
+		// Disconnect and disable all affected wallets.
+		for _, wallet := range affectedWallets {
+			if wallet.connected() {
+				wallet.Disconnect() // before disable or it refuses
+			}
+			wallet.setDisabled(true)
 		}
```

Comment:
Might not matter, but we should probably disconnect and disable the parent wallet last.